### PR TITLE
Development

### PR DESCRIPTION
**Pull request by hakichris**
In this branch, we developed the feature and functionality of the budget app 

- [x] create landing page for user to sign in and register
- [x]  create a welcome page for everyone 
- [x] set authentication for users to sign and register with full name, email and password 
- [x] create the category page for users to list them 
- [x] add a create category button to create new category from the user account 
- [x] create transaction page from category 
- [x] list transactions belonging to each category 
- [x] create transaction page for adding new transaction 
- [x] add styling and back button for user navigation. 

Kindly find the link to the video presentation :    https://drive.google.com/file/d/1EFDXr170mEEKHK4AsqTws01FuRJRN8LQ/view?usp=sharing

### CODE REVIEW

File: app/models/category.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class Category < ApplicationRecord
+  belongs_to :user
+  has_one_attached :icon
+  has_many :cat_transacts, dependent: :destroy
+  has_many :financial_transactions, through: :cat_transacts
+
+  validates :name, presence: true, length: { maximum: 250 }
+
+  def total(category_id)
+    financial_transactions.joins(:cat_transacts).where(cat_transacts: { category_id: }).sum('financial_transactions.amount')
```
Comment (CollinsTatang) [inline app/models/category.rb:10]: - [ ] We have a syntax error here in line 10 that cost our app not to run smoothly 
- [ ] Your `total method` is not working as expected, `the total should return the sum of all transactions NOT the amount of just one transaction`
version

**Review by CollinsTatang — CHANGES_REQUESTED** [review]
Hi @hakichris , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

## Required changes 

- [ ] Please check out your styling files, your app has no style on your live demo and localhost but I see the style on the video presentation on your end.

It's ok to have bugs in your code. Please, see this code review as a chance for you to learn practical coding skills and learn new tips that can improve your code quality and help you get a better job.
_Check the comments under the review._
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/category.rb (hunk 1 above)
Comment (hakichris) [reply app/models/category.rb:10]: Hello @CollinsTatang, thank you for this. It is fixed. per requirements it is returning the total sum of all transactions that belongs to one category. you can create multiple transactions in one category and check it. it is working fine on my end. Thank you.

Comment (CollinsTatang) [reply app/models/category.rb:10]: Thank you @hakichris 
I have to check it out and it's working well expected

- [ ] All the styling load well and meet the design on mobile

**Comment by arslanbisharat** [comment]
Hi @contributor_name,
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/category.rb (hunk 1 above)
Comment (CollinsTatang) [reply app/models/category.rb:10]: Hi @hakichris 
After extra review from co-colleague 
- I notice that the line 10 error was due to my Ruby Version 2.7.0, I will be updating my Ruby version asap.
Thank you for reaching out @hakichris

**Pull request merged**